resources/components/ImageCard.vue

```diff
@@ -68,6 +88,12 @@ module.exports = {
 	methods: {
 		toggleConfirmed: function ( suggestion ) {
 			suggestion.confirmed = !suggestion.confirmed;
+		},
+		onPublish: function () {
+			console.log( 'On publish' );
+		},
+		onSkip: function () {
```

Comment:
I think we may have removed the array of current images from state too soon; it would probably be better to dispatch an action here vs. emitting something up to the parent. What do you think @egardner?

Reply:
Yes, this should be a Vuex action. My thinking was that the ImageCard would dispatch an action on publish which knows 1) what image is under consideration and 2) what tags the user has approved. These things can be included in a "payload" (similar to how you can do with events) that comes along for the ride. I'll look at this again when I build out the publish action.

Reply:
Right, I'm more concerned about the skip button here. Right now the cardstack component knows about the list of images and the current image, but the skip action needs to be dispatched from the current image card.

Reply:
I have a basic `skipImage` action [defined](https://github.com/egardner/mediawiki-extensions-MachineVision/blob/master/resources/store/index.js#L268) in the Vuex store so far. We can use Vuex's `mapActions` helper to make that action accessible as a method inside of ImageCard.

The thinking was, skipping removes the foremost image from the Vuex queue (we could also add an API call here if we need to inform the server that skipping has happened); since the queue of images has now changed, everything else should re-render with updated data. If Vue keeps the ImageCard component in place and just passes new props down to it, we can add a `watch` function to clear out the existing suggestions that get copied into ImageCard's state. But this might not even be necessary.

Anything I'm missing here?

Reply:
That all makes sense 👍